### ISSUE TITLE
richer file ignores for image builds

### ISSUE THREAD

**Issue by nicks** [windmilleng/tilt#1656]
Many Tilt users develop in repos with multiple image builds.

Tilt can be very clunky for them, because a single change forces all the images to rebuild. They need each image build to ignore large parts of the repo.

We're planning to add an `ignore` argument to docker_build, to tell Tilt to ignore files on a per-image basis.

Here's a longer write-up of the motivation for this feature and possible alternative implementations:
https://github.com/windmilleng/tilt.specs/blob/master/slim_build_context.md

**Comment by nicks** [comment]
this also came up in https://github.com/windmilleng/tilt/issues/1606 and a few meetings with individual teams

**Comment by samscott89** [comment]
To add to this, I would love it if Tilt could be configured to respect `.dockerignore` and/or `.gitignore` files as well.

**Comment by jazzdan** [comment]
Two changes (https://github.com/windmilleng/tilt/commit/69381c6db2209f6ceff549cebf67319c58ceff48, https://github.com/windmilleng/tilt/commit/31766e703cf2e7f6909db871458e04b376192d66) were recently made that add two parameters to `docker_ignore` to help address this: `ignore` and `only`. Here's the relevant snippet from the [as of yet to be deployed docs](https://github.com/windmilleng/tilt.build/pull/158/files):

```python
def docker_build(ref: str, context: str, ignore: Union[str, List[str]] = [], only: Union[str, List[str]] = []) -> None:
  """Args:

    ...snip...
    
    ignore: set of file patterns that will be ignored. Ignored files will not trigger builds and will not be included in images. Follows the `dockerignore syntax <https://docs.docker.com/engine/reference/builder/#dockerignore-file>`_.
    only: set of file patterns that will cause a build to be triggered. All other changes will be ignored. Inverse of ignore parameter. Follows the `dockerignore syntax <https://docs.docker.com/engine/reference/builder/#dockerignore-file>`_. Note that ignores take precedence over this parameter.
  """
```

@samscott89 does this look like it would address your issue? If you get a chance to try it out on master we'd love the feedback!

**Comment by samscott89** [comment]
Hey @jazzdan! This looks awesome. To be honest, looking back to my original request, I think it was caused by a misunderstanding of how to use `docker_build` and `watch_file`.

I think I was originally specifying `watch_file` for, e.g. `watch_file('src/')` along with `docker_build('tag', 'src/')`. But this is effectively redundant, right? I.e. `docker_build` is already looking for changes in `src/`, and only those which aren't ignored by `.dockerignore`?

Basically, my desire was that `watch_file` could be configured to also ignore files specified by docker/git ignores. But this is for a use case which I don't think I actually need.

**Comment by jazzdan** [comment]
Thanks @samscott89! Yeah that's right, `docker_build` automatically watches the context.

Closing per above. Let us know if you have any other questions/feature requests!
